### PR TITLE
Feat/slim tokens

### PR DESCRIPTION
**Pull request by giovannic**
## Summary

Changes tokens to match giovannic:feat/linear_attention. Inline with the [0.1.0 roadmap](https://github.com/mrc-ide/tfmpe/wiki/Route-to-version-0.1.0). This is the rationale for the changes to Tokens:

 1. They are more suitable for an encoder only Transformer architecture. All tokens (conditional or not) are passed into a single encoder.
 2. They no longer use self-attention masks or Independence The gains from specifying independence were detrimental to FMPE accuracy and comes at a performance cost for linear attention.
 3. They simplify token logic. No carrying around labelling/independence metadata, all tokens are computed eagerly and can be manipulated with `jax.tree.map` or `dataclasses.replace`.
 4. No silly TokenView stuff

### New Changes

 * Labeller class to enforce consistent labelling between Token objects

### Breaking Changes

 * Tokens must be initialised with conditional and target data
 * Tokens must specify a condition argument
 * Tokens no longer accept "Independence"
 * Users can no longer construct "Independence"
 * No more TokenView
---

### All PRs
- [ ] **Correctness**: Logic is sound and implementation is correct
- [ ] **Testing**: Appropriate tests added/updated
- [ ] **Benchmarking**: Training/sampling speed impact assessed if relevant
- [ ] **Documentation**: Code comments and docs updated where needed

### CODE REVIEW

**Review by M-Kusumgar — APPROVED** [review]
nice, this looks simpler for sure, i checked out the tests for how labeller is used, seems sensible, its basically a metadata class for how to read these tokens?
